### PR TITLE
Update dev deps not part of build and actions

### PR DESCRIPTION
**Pull request by jfranciscosousa**
Updates all devs deps not part of the build process, meaning everything except `rollup` stuff as it as a lot of breaking changes. I'll take a look later!

Also updates the Github actions to the latest major versions available.
AND changes the NodeJS matrix to use node `14` and `18` as well.

### CODE REVIEW

**Comment by jfranciscosousa** [comment]
The changes on the test it's because you cannot reassign location stuff on the newest jsdom version, we have to use the `global` package and use `jsdom.reconfigure` for it.

File: src/test/helpers.test.ts [hunk 1]
```diff
@@ -94,7 +92,7 @@ describe('test decodedUrlParams', () => {
 
 describe('test isModalOpen', () => {
   afterEach(() => {
-    global.window.location = realLocation;
+    //global.window.location = realLocation;
```
Comment (donaminos) [inline src/test/helpers.test.ts:95]: Why adding and commenting this ?

Comment (jfranciscosousa) [reply src/test/helpers.test.ts:95]: Leftovers from debugging. Will remove

Comment (donaminos) [reply src/test/helpers.test.ts]: Looks good to me 👍

**Pull request merged**